rust/libnewsboat-ffi/src/matchererror.rs

```diff
@@ -0,0 +1,64 @@
+use abort_on_panic;
+use libc::c_char;
+use libnewsboat::matchererror::MatcherError;
+use std::ffi::CString;
+use std::ptr;
+
+// These constants MUST match numbers in `enum MatcherException::Type`, see include/matcherexception.h
+const ATTRIB_UNAVAIL: u8 = 0;
+const INVALID_REGEX: u8 = 1;
+
+#[repr(C)]
+pub struct MatcherErrorFfi {
+    err_type: u8, // One of the constants defined above
+    info: *mut c_char,
+    info2: *mut c_char,
```

Comment:
I think it would be nice to just save the formatted `what` message instead of somewhat arbitrary `info` and `info2`.

(just nice in the future, not something I expect to get changed in this PR)

Reply:
Yeah, I just followed our existing C++ code here. Not sure why it was written like this. Perhaps the author prepared for some future extensions like multiple ways to display the error?

I'm fine with having this changed, but I don't think it's even worth filing an issue. I don't plan to change it myself.